base/some.jl

```diff
@@ -145,3 +145,7 @@ macro something(args...)
     something = GlobalRef(Base, :something)
     return :($something($expr))
 end
+
+==(a::Some, b::Some) = a.value == b.value
+isequal(a::Some, b::Some)::Bool = isequal(a.value, b.value)
+hash(s::Some, h::UInt) = hash(s.value, hash(Some, h))
```

Comment:
```suggestion
const hash_some_seed = UInt == UInt64 ? 0xde5c997007a4ca3a : 0x78c29c09
hash(s::Some, h::UInt) = hash(s.value, hash_some_seed + h)
```

It's better to use a random seed to avoid this hash collision

```julia
julia> hash((Some(1),)) == hash((1, Some))
true
```

And it's faster

```julia
julia> using Chairmarks

julia> Base.hash(s::Some, h::UInt) = hash(s.value, hash(Some, h))

julia> @b Some(rand()) hash
5.933 ns

julia> Base.hash(s::Some, h::UInt) = hash(s.value, hash_some_seed + h)

julia> @b Some(rand()) hash
2.836 ns
```